### ISSUE TITLE
stopping video encoder with surface crash on emulator

### ISSUE THREAD

**Issue by troy-lamerton** [pedroSG94/rtmp-rtsp-stream-client-java#482]
On Bluestacks emulator, stopping the video encoder causes crash due to SIGSEGV on 'glThread'. Cannot reproduce on real phone, only crashes on bluestacks.

Full logcat:
https://gist.github.com/troy-lamerton/c9b937d5c8c9afd29bb861968ed35fbd

- Can always reproduce crash in the OpenGl RTMP activity.
- Cannot reproduce in the Custom RTMP activity.

Perhaps the key difference is encoder using inputSurface (instead of buffers like in RtmpCamera1).

https://github.com/thesingularitygroup/rtmp-rtsp-stream-client-java/blob/b683775f13b15c17aac310068e23cbfe7f0b7464/rtplibrary/src/main/java/com/pedro/rtplibrary/base/Camera1Base.java#L189

The crash happens almost immediately after `removeMediaCodecSurface` which releases an egl context: https://github.com/thesingularitygroup/rtmp-rtsp-stream-client-java/blob/53ea378599b8205c19355bf7c27e453572ae8c58/encoder/src/main/java/com/pedro/encoder/input/gl/SurfaceManager.java#L137

Extra notes:

When the SIGSEGV is logged, the activity exits and the menu activity reappears.
Same behaviour when doing start/stop record vs start/stop stream.

Not sure how to proceed? Let me know what I can do to help find the cause.

**Comment by pedroSG94** [comment]
This seems crash here:
https://github.com/thesingularitygroup/rtmp-rtsp-stream-client-java/blob/53ea378599b8205c19355bf7c27e453572ae8c58/encoder/src/main/java/com/pedro/encoder/input/gl/SurfaceManager.java#L141
This is a OpenGl crash so I don't know how to fix it. I need investigate about it. I will try reproduce with that emulator.

**Comment by troy-lamerton** [comment]
Can I help with anything? Let me know if you're blocked because we might be able to fix it internally.

**Comment by pedroSG94** [comment]
Sorry for late response. I had problems installing emulator because normally I use Linux and this emulator is only for windows or mac and I had no time this week. I will notify you tomorrow or next day with my tests results.

**Comment by pedroSG94** [comment]
I was doing tests with blustacks emulator but I got this results:
I opened app in OpenGlRtmp activity but with black screen. I dont know how to use camera in emulator.
I tried press record button and in logs seems that open camera and start encoder but any frame is renderer. No crash in opengl when stop (maybe because opengl never render anything).
I can't open webcam camera with my app I also tried with others apps in app store
